drivers/timer/arcv2_timer0.c

```diff
@@ -235,7 +234,7 @@ void z_clock_set_timeout(s32_t ticks, bool idle)
 	 * that interrupts are already disabled)
 	 */
 #ifdef CONFIG_SMP
-	if (IS_ENABLED(CONFIG_TICKLESS_IDLE) && idle && ticks == K_FOREVER) {
+	if (IS_ENABLED(CONFIG_TICKLESS_IDLE) && ticks == K_FOREVER) {
```

Comment:
Could you please elaborate a bit more what was wrong here in the first place?
Otherwise how can we justify exactly the change you made?

Reply:
For SMP, as we have gfrc, so no need to rely on idle.  What's more, if ticks ==K_FOREVER (-1) and idle == false,  then following codes can not work correctly

Reply:
1. How SMP & GFRC usage exclude a need of `idle`? It's not that I don't like that change but instead I'd just like to really understand what is done and why. Please pardon my noob questions but I'm sure that will simplify understanding rationale for your changes for other readers who may have less experience with Zephyr than you.

2. Whatever justification you have please put it in the commit message so via `git log` or `git blame` people may figure out what was changed and why.

Reply:
already add extra comments in commit msg and code comments